src/fieldViews/ProseMirrorFieldView.ts

```diff
@@ -0,0 +1,253 @@
+import type { Node, Schema } from "prosemirror-model";
+import { DOMParser, DOMSerializer } from "prosemirror-model";
+import type { Plugin, Transaction } from "prosemirror-state";
+import { EditorState } from "prosemirror-state";
+import { Mapping, StepMap } from "prosemirror-transform";
+import type { Decoration } from "prosemirror-view";
+import { DecorationSet, EditorView } from "prosemirror-view";
+import type { FieldView } from "./FieldView";
+import { FieldType } from "./FieldView";
+
+/**
+ * A FieldView that represents a
+ * nested rich text editor interface.
+ */
+export abstract class ProseMirrorFieldView<LocalSchema extends Schema = Schema>
+  implements FieldView<string> {
+  public static fieldType = FieldType.CONTENT;
+  public static defaultValue = "";
+
+  // The parent DOM element for this view. Public
+  // so it can be mounted by consuming elements.
+  public fieldViewElement = document.createElement("div");
+  // The editor view for this FieldView.
+  private innerEditorView: EditorView | undefined;
+  // The decorations that apply to this FieldView.
+  private decorations = new DecorationSet();
+  // The serialiser for the Node.
+  private serialiser: DOMSerializer;
+  // The parser for the Node.
+  private parser: DOMParser;
+
+  constructor(
+    // The node that this FieldView is responsible for rendering.
+    private node: Node,
+    // The outer editor instance. Updated from within this class when the inner state changes.
+    private outerView: EditorView,
+    // Returns the current position of the parent FieldView in the document.
+    private getPos: () => number,
+    // The offset of this node relative to its parent FieldView.
+    private offset: number,
+    // The schema that the internal editor should use.
+    schema: LocalSchema,
+    // The initial decorations for the FieldView.
+    decorations: DecorationSet | Decoration[],
+    // The ProseMirror node type name
+    private readonly propName: string,
+    // Plugins that the editor should use
+    private plugins?: Plugin[]
```

Comment:
Possible nit: if we don't use plugins outside of the construction of the editor, let's pass this into `createInnerEditorView` as an argument, instead, to keep them scoped to where they're used.